cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MdsKey.java

```diff
@@ -17,9 +17,14 @@
 package co.cask.cdap.data2.metadata.dataset;
```

Comment:
nope, we shouldn't be using class names for figuring out the type of the entity in this class. I will sync up with @knazir today. I think we should do https://github.com/caskdata/cdap/pull/6646/files#r78016553. This class, as well as the changes in ``KeyHelper`` are not backwards compatible.